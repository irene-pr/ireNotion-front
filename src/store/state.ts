const state = {
  isLoggedIn: false,
  userContent: {
    name: "",
    boards: [],
  },
  userData: {
    userId: "",
    username: "",
  },
  themeHeaders: "night-mode",
  themeSurfaces: "day-mode",
  isBoardEditModal: false,
  isUpdateNoteModal: false,
  idForModal: "",
};

export default state;
