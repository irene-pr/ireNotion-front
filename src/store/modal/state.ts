import { IModalState } from "@/types/store";

const state = (): IModalState => ({
  isBoardEditModal: false,
  isUpdateNoteModal: false,
  idForModal: "",
});

export default state;
