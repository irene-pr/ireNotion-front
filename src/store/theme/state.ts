import { IThemeState } from "@/types/store";

const state = (): IThemeState => ({
  themeHeaders: "night-mode",
  themeSurfaces: "day-mode",
});

export default state;
