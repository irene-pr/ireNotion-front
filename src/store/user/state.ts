import { IUserState } from "@/types/store";

const state = (): IUserState => ({
  isLoggedIn: false,
  userContent: {
    name: "",
    boards: [],
  },
  userData: {
    userId: "",
    username: "",
  },
  isLoading: false,
});

export default state;
