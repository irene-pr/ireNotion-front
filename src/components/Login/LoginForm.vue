<template>
  <div class="login-container">
    <router-link :to="paths.home">
      <h1 class="login-container__main-heading">ire<span>N</span>otion</h1>
    </router-link>
    <h2 class="login-container__secondary-heading">Welcome to ireNotion</h2>

    <form class="login-form" @submit.prevent="onSubmit" autocomplete="off">
      <div class="login-form__input-container">
        <label class="login-form__label" for="username">Username</label>
        <input
          type="text"
          id="username"
          class="login-form__input"
          v-model="username"
          placeholder="mario123"
        />
        <label class="login-form__label" for="password">Password</label>
        <input
          type="password"
          id="password"
          class="login-form__input"
          v-model="password"
          placeholder="at least 7 characters"
        />
      </div>
      <p class="message">
        {{ isMessageShown ? "Wrong username or password. Try again" : "" }}
      </p>
      <button
        class="login-form__button login-form__button--login"
        type="submit"
        :disabled="username === '' || password === ''"
      >
        LOG <span>IN</span>
      </button>
    </form>

    <div class="divider">
      <div class="divider__line divider__line--first"></div>
      <p class="divider__or">or</p>
      <div class="divider__line divider__line--second"></div>
    </div>
    <button class="login-form__button login-form__button--signup">
      SIGN <span>UP</span>
    </button>
  </div>
</template>

<script lang="ts">
import { defineComponent } from "vue";
import { mapActions } from "vuex";
import { UserLoginData } from "@/types/interfaces";
import paths from "../../router/paths";
import router from "@/router";

export default defineComponent({
  name: "LoginForm",
  components: {},
  data() {
    return {
      paths,
      username: "",
      password: "",
      isMessageShown: false,
    };
  },
  methods: {
    ...mapActions(["loginUser"]),
    async onSubmit() {
      if (this.username !== "" && this.password !== "") {
        const userLoginData: UserLoginData = {
          username: this.username,
          password: this.password,
        };
        try {
          await this.loginUser(userLoginData);
          router.push(paths.userBoard);
        } catch {
          this.isMessageShown = true;
        }
      }
    },
  },
});
</script>

<style lang="scss">
@import "@/styles/_variables";
@import "@/styles/_extends";

.login-container {
  width: 380px;
  padding-bottom: 20px;

  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: $theme-light-color;
  color: $theme-dark-color;
  font-family: $Lato;

  a {
    text-decoration: none;
  }
  &__main-heading {
    font-family: $h1;
    font-size: 40px;
    font-weight: 100;
    color: $theme-dark-color;
    span {
      color: $theme-orange;
    }
  }
  &__secondary-heading {
    color: $theme-dark-color-opacity;
    font-family: $h2;
    font-size: 20px;
    font-weight: 500;
    letter-spacing: 0.03em;
  }
  @media only screen and (max-width: 480px) {
    height: 100%;
    width: 100%;
    justify-content: center;
  }
}
.login-form {
  padding: 30px 0 10px 0;
  display: flex;
  flex-direction: column;
  align-items: center;

  &__input-container {
    display: flex;
    flex-direction: column;
  }

  &__label {
    padding: 5px 0 5px 2px;
    @extend %label-text;
  }
  &__input {
    height: 30px;
    padding-bottom: 0;
    border: none;
    border-bottom: 1px solid $theme-form-line;
    background-color: transparent;
    color: $theme-dark-color;
    font-size: 14px;
    &::placeholder {
      font-family: $Lato;
      opacity: 0.85;
      letter-spacing: 0;
    }
    &[type="password"] {
      font-family: Verdana, serif;
      letter-spacing: 0.125em;
    }
    &:focus {
      outline: none;
      color: $theme-orange;
    }
  }
  &__button {
    @extend %button;
    position: initial;
    &--login {
      @extend %button--login;
    }
    &--signup {
      @extend %button--signup;
      margin: 12px 0;
    }
    &:disabled {
      opacity: 0.6;
      cursor: initial;
    }
    span {
      @extend %button--span;
      color: $theme-orange;
    }
  }
}
.message {
  height: 15px;
  margin: 20px 0;
  @extend %label-text;
  font-size: 12px;
  color: $theme-orange;
}
.divider {
  width: 40%;
  display: flex;
  justify-content: space-around;
  align-items: center;
  &__or {
    @extend %label-text;
  }
  &__line {
    width: 50px;
    height: 0.1px;
    border-bottom: 1px solid $theme-form-line;
  }
}
</style>
