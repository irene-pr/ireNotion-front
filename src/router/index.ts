import { createRouter, createWebHistory, RouteRecordRaw } from "vue-router";
import paths from "@/router/paths";
import Home from "../views/Home.vue";
import Login from "../views/Login.vue";
import Register from "../views/Register.vue";
import BoardPage from "../views/BoardPage.vue";

const routes: Array<RouteRecordRaw> = [
  {
    path: paths.home,
    name: "Home",
    component: Home,
  },
  {
    path: paths.login,
    name: "Login",
    component: Login,
  },
  {
    path: paths.register,
    name: "Register",
    component: Register,
  },
  {
    path: paths.userBoard,
    name: "BoardPage",
    component: BoardPage,
  },
];

const router = createRouter({
  history: createWebHistory(process.env.BASE_URL),
  routes,
});

export default router;
