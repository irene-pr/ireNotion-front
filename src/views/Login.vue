<template>
  <main class="login-page">
    <LoginForm class="login-page__form-component" />
  </main>
</template>

<script lang="ts">
import { defineComponent } from "vue";
import { mapActions, mapGetters, mapState } from "vuex";
import LoginForm from "@/components/Login/LoginForm.vue";

export default defineComponent({
  name: "Login",
  components: {
    LoginForm,
  },
  computed: {
    ...mapState(["isLoggedIn"]),
  },
  methods: {
    ...mapActions(["checkToken"]),
    ...mapGetters(["redirectToUserBoard"]),
  },
  /* mounted() {
    this.checkToken();
    if (this.isLoggedIn) {
      this.redirectToUserBoard();
    }
  }, */
});
</script>

<style scoped lang="scss">
@import "@/styles/_variables";

.login-page {
  height: 100vh;
  background-image: url("../assets/background-2.webp");
  background-size: cover;
  display: flex;
  justify-content: center;
  align-items: center;
}
</style>
