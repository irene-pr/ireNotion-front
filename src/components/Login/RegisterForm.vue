<template>
  <div class="register-container">
    <router-link :to="paths.home">
      <h1 class="register-container__main-heading">ire<span>N</span>otion</h1>
    </router-link>

    <form class="register-form" @submit.prevent="onSubmit" autocomplete="off">
      <div class="register-form__input-container">
        <label class="register-form__label" for="name">Name</label>
        <input
          type="text"
          id="name"
          class="register-form__input"
          v-model="name"
          placeholder="Mario"
        />
        <label class="register-form__label" for="username">Username</label>
        <input
          type="text"
          id="username"
          class="register-form__input"
          v-model="username"
          placeholder="mario123"
        />
        <label class="register-form__label" for="password">Password</label>
        <input
          type="password"
          id="password"
          class="register-form__input"
          v-model="password"
          placeholder="at least 7 characters"
        />
        <label class="register-form__label" for="repeat-password"
          >Repeat password</label
        >
        <input
          type="password"
          id="repeat-password"
          class="register-form__input"
          v-model="repeatPassword"
          placeholder="at least 7 characters"
        />
      </div>
      <p class="message">
        {{ messageShown }}
      </p>
      <button
        class="register-form__button register-form__button--signup"
        type="submit"
        :disabled="
          name === '' ||
          username === '' ||
          password === '' ||
          repeatPassword === ''
        "
      >
        SIGN <span>UP</span>
      </button>
    </form>

    <div class="divider">
      <div class="divider__line divider__line--first"></div>
      <p class="divider__or">or</p>
      <div class="divider__line divider__line--second"></div>
    </div>
    <router-link :to="paths.login">
      <button class="register-form__button register-form__button--login">
        LOG <span>IN</span>
      </button>
    </router-link>
  </div>
</template>

<script lang="ts">
import { defineComponent } from "vue";
import { mapActions } from "vuex";
import { IUserRegisterData } from "@/types/interfaces";
import paths from "../../router/paths";
import router from "@/router";

export default defineComponent({
  name: "RegisterForm",
  components: {},
  data() {
    return {
      paths,
      name: "",
      username: "",
      password: "",
      repeatPassword: "",
      messageShown: "",
    };
  },
  methods: {
    ...mapActions(["registerUser"]),
    async onSubmit() {
      if (
        this.name !== "" &&
        this.username !== "" &&
        this.password !== "" &&
        this.repeatPassword !== "" &&
        this.password === this.repeatPassword &&
        this.password.length >= 7 &&
        this.password.length <= 20
      ) {
        const userRegisterData: IUserRegisterData = {
          name: this.name,
          username: this.username,
          password: this.password,
        };

        await this.registerUser(userRegisterData);
        router.push(paths.login);
      } else if (this.password !== this.repeatPassword) {
        this.messageShown = "The passwords don't match. Try again";
      } else if (this.password.length < 7 || this.password.length > 20) {
        this.messageShown =
          "The password must have between 7 and 20 characters";
      }
    },
  },
});
</script>

<style lang="scss" scoped>
@import "@/styles/_variables";
@import "@/styles/_extends";

.register-container {
  width: 380px;
  padding-bottom: 20px;

  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: $theme-light-color;
  color: $theme-dark-color;
  font-family: $Lato;

  a {
    text-decoration: none;
  }
  &__main-heading {
    font-family: $h1;
    font-size: 40px;
    font-weight: 100;
    color: $theme-dark-color;
    span {
      color: $theme-blue;
    }
  }
  &__secondary-heading {
    color: $theme-dark-color-opacity;
    font-family: $h2;
    font-size: 20px;
    font-weight: 500;
    letter-spacing: 0.03em;
  }
  @media only screen and (max-width: 480px) {
    height: 100%;
    width: 100%;
    justify-content: center;
  }
}
.register-form {
  padding: 30px 0 10px 0;
  display: flex;
  flex-direction: column;
  align-items: center;

  &__input-container {
    display: flex;
    flex-direction: column;
  }

  &__label {
    padding: 5px 0 5px 2px;
    @extend %label-text;
  }
  &__input {
    height: 30px;
    padding-bottom: 0;
    border: none;
    border-bottom: 1px solid $theme-form-line;
    background-color: transparent;
    color: $theme-dark-color;
    font-size: 14px;
    &::placeholder {
      font-family: $Lato;
      opacity: 0.85;
      letter-spacing: 0;
    }
    &[type="password"] {
      font-family: Verdana, serif;
      letter-spacing: 0.125em;
    }
    &:focus {
      outline: none;
      color: $theme-blue;
    }
  }
  &__button {
    @extend %button;
    position: initial;
    &--login {
      @extend %button--login;
      margin: 12px 0;
    }
    &--signup {
      @extend %button--signup;
    }
    &:disabled {
      opacity: 0.6;
      cursor: initial;
    }
    span {
      @extend %button--span;
      color: $theme-blue;
    }
  }
}
.message {
  height: 15px;
  margin: 20px 0;
  @extend %label-text;
  font-size: 12px;
  color: $theme-blue;
}
.divider {
  width: 40%;
  display: flex;
  justify-content: space-around;
  align-items: center;
  &__or {
    @extend %label-text;
  }
  &__line {
    width: 50px;
    height: 0.1px;
    border-bottom: 1px solid $theme-form-line;
  }
}
</style>
