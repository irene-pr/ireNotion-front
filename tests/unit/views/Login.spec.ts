import { mount } from "@vue/test-utils";
import { createStore } from "vuex";
import Login from "../../../src/views/Login.vue";
import state from "@/store/state";
import router from "@/router";
import LoginForm from "@/components/Login/LoginForm.vue";

describe("Given a Login view", () => {
  const store = createStore({
    state() {
      return state;
    },
    actions: {
      checkToken: jest.fn(),
    },
    getters: {
      redirectToUserBoard: jest.fn(),
    },
  });

  describe("When it is rendered", () => {
    test("Then it renders", () => {
      mount(Login, {
        global: {
          plugins: [router, store],
        },
        stubs: ["router-view"],
      });
    });
    test("Then the it matches the snapshot", () => {
      const wrapper = mount(Login, {
        global: {
          plugins: [router, store],
        },
        stubs: ["router-view"],
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    test("Then it renders the login form", () => {
      const wrapper = mount(Login, {
        global: {
          plugins: [router, store],
        },
        stubs: ["router-view"],
      });

      const loginForm = wrapper.getComponent(LoginForm);

      expect(loginForm).toBeTruthy();
    });
  });
  describe("When the isLoggedIn state is true", () => {
    test("Then it invokes the redirectToUserBoard getter", () => {
      state.isLoggedIn = true;

      mount(Login, {
        global: {
          plugins: [router, store],
        },
        stubs: ["router-link", "router-view"],
      });
    });
  });
});
