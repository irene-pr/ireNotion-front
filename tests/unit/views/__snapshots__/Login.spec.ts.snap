// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Login view When it is rendered Then the it matches the snapshot 1`] = `
<main
  class="login-page"
>
  <div
    class="login-container login-page__form-component"
  >
    <a
      class=""
      href="/"
    >
      <h1
        class="login-container__main-heading"
      >
        ire
        <span>
          N
        </span>
        otion
      </h1>
    </a>
    <h2
      class="login-container__secondary-heading"
    >
       Welcome back to ireNotion 
    </h2>
    <form
      autocomplete="off"
      class="login-form"
    >
      <div
        class="login-form__input-container"
      >
        <label
          class="login-form__label"
          for="username"
        >
          Username
        </label>
        <input
          class="login-form__input"
          id="username"
          placeholder="mario123"
          type="text"
        />
        <label
          class="login-form__label"
          for="password"
        >
          Password
        </label>
        <input
          class="login-form__input"
          id="password"
          placeholder="at least 7 characters"
          type="password"
        />
      </div>
      <p
        class="message"
      />
      <button
        class="login-form__button login-form__button--login"
        disabled=""
        type="submit"
      >
         LOG 
        <span>
          IN
        </span>
      </button>
    </form>
    <div
      class="divider"
    >
      <div
        class="divider__line divider__line--first"
      />
      <p
        class="divider__or"
      >
        or
      </p>
      <div
        class="divider__line divider__line--second"
      />
    </div>
    <a
      class=""
      href="/register"
    >
      <button
        class="login-form__button login-form__button--signup"
      >
         SIGN 
        <span>
          UP
        </span>
      </button>
    </a>
  </div>
</main>
`;
